embedders/classification/contextual.py

```diff
@@ -121,11 +122,11 @@ def _encode(
             documents_batch = [doc.replace("\n", " ") for doc in documents_batch]
             try:
                 if self.use_azure:
-                    response = openai.Embedding.create(
+                    response = embedding(
                         input=documents_batch, engine=self.model_name
                     )
                 else:
-                    response = openai.Embedding.create(
+                    response = embedding(
                         input=documents_batch, model=self.model_name
```

Comment:
in order to use cohere/replicate/other embedding models, just pass in the the `model` here